modules/misc/nix.nix

```diff
@@ -98,6 +98,31 @@ in {
       '';
     };
 
+    nixPath = mkOption {
+      type = types.listOf types.str;
+      default = [ ];
+      example = [
+        "$HOME/.nix-defexpr/channels"
+        "darwin-config=$HOME/.config/nixpkgs/darwin-configuration.nix"
+      ];
+      description = ''
+        Adds new directories to the Nix expression search path.
+
+        Used by Nix when looking up paths in angular brackets
+        (e.g. <literal>&lt;nixpkgs&gt;</literal>).
+      '';
+    };
+
+    keepOldNixPath = mkOption {
```

Comment:
instead of yet another config, allow nixPath to be null ?

Reply:
The main reason for this setting is, when you set `nix.nixPaths` whether you want it to keep the existing value of the environment variable or to fully override it

Reply:
I think this is unnecessary complexity. Always overriding when `nixPath` is non-null should be enough.

Reply:
As a user, I think it makes sense to want to be able to keep the old `NIX_PATH`, and prepend values that I want to override, rather than completely nuke the system-provided variable.

Reply:
Note that you can add stuff to the Nix path via the nix.conf file:

```
extra-nix-path = foo=/bar
```

Reply:
Unfortunately `NIX_PATH` takes precedence over whatever is set in `nix.conf(5)`, see NixOS/nix#7857.

Therefore as a user, if I want to be able to use some of the system's NIX_PATH entries, while overriding some of its values, then the proposed solution is the only one that works.